test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,359 @@
+package seedu.addressbook.data;
+
+import static org.junit.Assert.*;
+
+import java.util.Iterator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.data.tag.UniqueTagList.DuplicateTagException;
+import seedu.addressbook.data.tag.UniqueTagList.TagNotFoundException;
+
+/**
+ * we check, in particular, for the two guarantees specified by the AddressBook class
+ *  - Every tag found in every person will also be found in the tag list.
+ *  - The tags in each person point to tag objects in the master list. (== equality)
+ */
+public class AddressBookTest {
+	
+	private Tag tagDead;
+	private Tag tagRepublican;
+	private Tag tagDemocrat;
+	private Tag tagIndependent;
+
+	private Person personBarack; // has tagDemocrat
+	private Person personGeorge; // has tagRepublican
+	private Person personBill;   // has tagDemocrat
+	private Person personMonroe; // has tagIndependent, tagDead
```

Comment:
In other words, code should not give subtle hints as to the period in which it was written or the person who wrote it.